AppCenter/AppCenter/Internals/Channel/MSChannelGroupDefault.m

```diff
@@ -23,8 +24,12 @@ - (instancetype)initWithIngestion:(nullable MSAppCenterIngestion *)ingestion {
     _logsDispatchQueue = serialQueue;
     _channels = [NSMutableArray<id<MSChannelUnitProtocol>> new];
     _delegates = [NSHashTable weakObjectsHashTable];
-    _ingestion = ingestion;
     _storage = [MSLogDBStorage new];
+    if (ingestion) {
+      _ingestion = ingestion;
+      _ingestion.authToken = [MSAuthTokenContext sharedInstance].authToken;
+    }
+    [[MSAuthTokenContext sharedInstance] addDelegate:self];
```

Comment:
:thinking: Is it worth adding the delegate if ingestion is nil?

Reply:
Since the token is not used by One Collector it would not be useful according to me.

Reply:
I agree with you guys but I think we don't need to logically couple auth token context with ingestion. These are independent operations even though token will be sent to Ingestion.

Reply:
Isn't that ` _ingestion.authToken = [MSAuthTokenContext sharedInstance].authToken;` logically coupled already? Either way not a blocker to me.

Reply:
Ingestion doesn't know about `MSAuthTokenContext`, and `MSAuthTokenContext` doesn't know about Ingestion. In my opinion, they are already decoupled.